### PR TITLE
DRILL-4996: Parquet Date auto-correction is not working in auto-parti…

### PR DESCRIPTION
**Pull request by vdiravka**
…tioned parquet files generated by drill-1.6

- Changed detection approach of corrupted date values for the case, when parquet files are generated by drill:
  the corruption status is determined by looking at the min/max values in the metadata;
- Appropriate refactoring of TestCorruptParquetDateCorrection.

### CODE REVIEW

File: exec/java-exec/src/test/java/org/apache/drill/exec/physical/impl/writer/TestCorruptParquetDateCorrection.java [hunk 1]
```diff
@@ -201,68 +203,34 @@ public void testCorrectDatesAndExceptionWhileParsingCreatedBy() throws Exception
...
-        testPlanMatchingPatterns(query, new String[]{"numFiles=1"}, null);
-
-        // read with a filter on the partition column
-        testBuilder()
-            .sqlQuery(query)
-            .unOrdered()
-            .baselineColumns("date_col")
-            .baselineValues(new DateTime(15334, 03, 17, 0, 0))
-            .go();
+        for (String table : new String[]{CORRUPTED_PARTITIONED_DATES_1_2_PATH, CORRUPTED_PARTITIONED_DATES_1_4_0_PATH}) {
```
Comment (chunhui-shi) [inline exec/java-exec/src/test/java/org/apache/drill/exec/physical/impl/writer/TestCorruptParquetDateCorrection.java]: Could you keep for block the same indent as original text?

File: exec/java-exec/src/test/java/org/apache/drill/exec/physical/impl/writer/TestCorruptParquetDateCorrection.java [hunk 2]
```diff
@@ -273,26 +241,28 @@ public void testReadPartitionedOnCorruptedDates_UserDisabledCorrection() throws
...
-        testPlanMatchingPatterns(query, new String[]{"numFiles=1"}, null);
-
-        // read with a filter on the partition column
-        testBuilder()
-            .sqlQuery(query)
-            .unOrdered()
-            .baselineColumns("date_col")
-            .baselineValues(new DateTime(1970, 1, 1, 0, 0))
-            .go();
+        for (String table : new String[]{CORRUPTED_PARTITIONED_DATES_1_2_PATH, CORRUPTED_PARTITIONED_DATES_1_4_0_PATH}) {
```
Comment (chunhui-shi) [inline exec/java-exec/src/test/java/org/apache/drill/exec/physical/impl/writer/TestCorruptParquetDateCorrection.java]: The same. We always keep indent the same.

File: exec/java-exec/src/test/java/org/apache/drill/exec/physical/impl/writer/TestCorruptParquetDateCorrection.java (hunk 1 above)
Comment (vdiravka) [inline exec/java-exec/src/test/java/org/apache/drill/exec/physical/impl/writer/TestCorruptParquetDateCorrection.java]: It is necessary to keep indent (211 line) after `try` block and then (212 line) after `for` block.

File: exec/java-exec/src/test/java/org/apache/drill/exec/physical/impl/writer/TestCorruptParquetDateCorrection.java (hunk 2 above)
Comment (vdiravka) [inline exec/java-exec/src/test/java/org/apache/drill/exec/physical/impl/writer/TestCorruptParquetDateCorrection.java]: Answered above.

**Comment by parthchandra** [comment]
+1

**Pull request closed without merging**